// For uploading diagnosis keys

syntax = "proto2";

package diagnosis;

option optimize_for = LITE_RUNTIME;

message SubmissionPayload {
  repeated TemporaryExposureKey keys = 1;
}

message TemporaryExposureKey {
  // Key of infected user
  optional bytes key_data = 1;

  // Varying risks associated with exposure depending on type of verification
  optional int32 transmission_risk_level = 2;

  // The interval number since epoch for which a key starts
  optional int32 rolling_start_interval_number = 3;

  // Increments of 10 minutes describing how long a key is valid
  optional int32 rolling_period = 4
      [default = 144];  // defaults to 24 hours
}

